R/calculate_within_batch_ari.R

```diff
@@ -48,54 +48,96 @@ calculate_within_batch_ari <- function(individual_sce_list,
   }
 
   # Check that `pc_name` is in merged SCE object
-  if (!pc_name %in% reducedDimNames(merged_sce)) {
-    stop("The provided `pc_name` cannot be found in the `merged_sce`.")
+  if (!any(pc_names %in% reducedDimNames(merged_sce))) {
+    stop("One or more of the PC names provided in `pc_names` cannot be found in the `merged_sce`.")
   }
 
-  # Pull out the PCs or analogous reduction from merged object
-  merged_pcs <- reducedDim(merged_sce, pc_name)
+  # Calculate within-batch ARI values across list of PCs
+  within_batch_ari_tibble <-
+    purrr::map(
+      pc_names,
+      \(pcs)
+      within_batch_ari_from_pcs(
+        merged_sce = merged_sce,
+        batch_column = batch_column,
+        pc_name = pcs
+      )
```

Comment:
Pass the `BLUSPARAM` to the internal function
```suggestion
        pc_name = pcs,
        BLUSPARAM = BLUSPARAM
      )
```